### DISCUSSION TITLE
Hombrew Fonts on Linux have Disappeared

### DISCUSSION THREAD

**Discussion by iandol** [Homebrew/discussions#5375] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
▶︎ brew config
HOMEBREW_VERSION: 4.3.0-40-g1fac683
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1fac68348a61f735066ba58e6ba35d2cec8bdf8c
Last commit: 30 hours ago
Core tap HEAD: 27d359a3789a19aad6d79aebfd5cb1e2d1daca71
Core tap last commit: 26 hours ago
Core tap JSON: 16 May 08:23 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :1
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit skylake
Clang: N/A
Git: 2.34.1 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 6.5.0-35-generic x86_64 GNU/Linux
OS: Ubuntu 22.04.4 LTS (jammy)
Host glibc: 2.35
/usr/bin/gcc: 11.4.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: 14.1.0
xorg: N/A
```

### Output of `brew doctor`

```shell
▶︎ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  font-fantasque-sans-mono
  font-alegreya-sans
  font-libertinus
  font-jetbrains-mono
  font-alegreya
  font-cascadia-code
  font-fira-code

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/LabJackM.h
  /usr/local/include/labjackusb.h

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libchipmunk.a

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  pkg-config
```

### Description of issue

We used to be able to use linuxbrew/fonts to install fonts with homebrew on Linux (which mirrored the macOS font tap; because fonts are installed as casks on macOS and casks are not supported on Linux, thus this difference in taps). linuxbrew/fonts was deprecated and the tap moved to homebrew/linux-fonts AFAICT.

Recently it seems homebrew have moved the macOS fonts into the main cask tap. The side effect of this is that that is now empty and the linux fonts have all disappeared:

https://github.com/homebrew/homebrew-cask-fonts/tree/0ed20f35d21ebc3f4f829689b101078718ad5ce1

```
brew tap homebrew/linux-fonts
brews search font-alegreya
```

returns nothing on Ubuntu but does on my macbook: https://formulae.brew.sh/cask/font-alegreya#default

You cann see this in `brew doctor`:

```
Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  font-fantasque-sans-mono
  font-alegreya-sans
  font-libertinus
  font-jetbrains-mono
  font-alegreya
  font-cascadia-code
  font-fira-code
```

What is the expected migration path for the use of brew for installing fonts on Linux? Thank you for all your hard work!

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-linux-fonts/issues/42

**Reply by iandol** [reply]
Ah, thanks for the details. That issue links to https://github.com/Homebrew/homebrew-cask/issues/112102#issuecomment-2092441493 where the details are clearly laid out.

@miccal and others completed this move, but it still isn't clear to me from reading that thread if this also means homebrew-linux-fonts is supposed to be tweaked to read from casks/fonts. This is not the case at present.

**Reply by iandol** [reply]
The action from homebrew-linux-fonts was here https://github.com/Homebrew/homebrew-linux-fonts/actions/runs/9120914160/workflow -- this is what I assume converted casks to formulae so that they were useable on Linux. I don't know how actions work, but I can't obviously see the source address there (is it line 31, that appears a cleanup)?

**Comment by iandol** [comment]
https://github.com/Homebrew/homebrew-linux-fonts/issues/42 is now closed after the migration (thanks). However I think the migration code for macOS is now stopping linux installing fonts from that repo, when you try it recommends to use the cask but casks are not supported on linux.

```bash
Warning: Formula homebrew/linux-fonts/font-inconsolata was renamed to homebrew/cask/font-inconsolata.
Warning: No available formula with the name "font-inconsolata". Did you mean font-inconsolata-g or font-inconsolata-lgc?
It was migrated from homebrew/linux-fonts to homebrew/cask.
You can access it again by running:
  brew tap homebrew/cask
And then you can install it by running:
  brew install --cask font-inconsolata
```

```bash
▶︎ brew tap
homebrew/core
homebrew/linux-fonts
rsteube/tap
```

**Comment by unikitty37** [comment]
I still can't see how to install fonts on Linux using homebrew.

When I type `brew tap homebrew/cask`, as suggested above, I get

```
Error: Tapping homebrew/cask is no longer typically necessary.
Add --force if you are sure you need it for contributing to Homebrew.
```

but if I don't force it and then type `brew search font-` it returns

- cfonts
- font-util
- fontconfig
- fontforge
- fonts-encodings
- fonttools
- libfontenc
- libxfont
- libxfont2
- mkfontscale
- fnt

and no actual fonts. (Arguably, only `font-util` should really have been returned out of those, since there's no hyphen after "font" in any of the other results 😛)

What do I need to do to actually be able to install fonts from Homebrew on Linux these days? i tried force-tapping homebrew/cask and that didn't return any fonts either, despite the warning @iandol got.

**Reply by SMillerDev** [reply]
You need to `brew tap homebrew/linux-fonts`. Just like before. Casks don't work on Linux.
